### PR TITLE
compat: Add subnet mask behind IP address to match Docker API

### PR DESCRIPTION
**Pull request by serverwentdown**
#### What this PR does / why we need it:

Mismatch between Podman and Docker APIs when listing network containers.

#### How to verify it

On both Docker and Podman, create a container on a network and run:
```
curl -X GET --unix-socket /var/run/docker.sock http://d/networks/<id> | jq
```

Docker (20.10.10):
Screenshot 2021-11-17 at 20 09 49
Podman:
Screenshot 2021-11-17 at 20-11-56

#### Which issue(s) this PR

Fixes: None

(Sidenote: I'm working on compatibility with `k3d`)

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@serverwentdown thanks for the PR.  Our CI requires a test to be added with a PR.  Could you create one or update one?  It would be good to test that the format is as you want it to be.

Changes LGTM otherwise, but I'm not sure if there are dependencies elsewhere for the current format.

File: pkg/api/handlers/compat/networks.go [hunk 1]
```diff
@@ -69,12 +70,14 @@ func convertLibpodNetworktoDockerNetwork(runtime *libpod.Runtime, network nettyp
 			return nil, err
 		}
 		if netData, ok := data.NetworkSettings.Networks[network.Name]; ok {
+			ipv4Address := fmt.Sprintf("%s/%d", netData.IPAddress, netData.IPPrefixLen)
+			ipv6Address := fmt.Sprintf("%s/%d", netData.GlobalIPv6Address, netData.GlobalIPv6PrefixLen)
```
Comment (Luap99) [inline pkg/api/handlers/compat/networks.go:74]: You have to check that `netData.IPAddress` is not an empty string before comping them to the CIDR. Right now you will return `/0` is there is no IP. It should just be empty.

**Review by Luap99 — CHANGES_REQUESTED** [review]
One nit, could you also add a test to https://github.com/containers/podman/tree/main/test/apiv2?

**Comment by serverwentdown** [comment]
@Luap99 I have added an APIv2 test but because the API tests run as rootless, it won't populate the IP address field.

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by Luap99** [comment]
Tanks @serverwentdown, unfortunately you have to rebase to the latest main branch in order to pass the CI tests.

**Comment by rhatdan** [comment]
Forcing merge to save the planet.

**Comment by rhatdan** [comment]
Thanks @serverwentdown

**Comment by serverwentdown** [comment]
@rhatdan @Luap99 is there any plans for this bugfix to be released in v3 or is the goal v4?

**Comment by Luap99** [comment]
> @rhatdan @Luap99 is there any plans for this bugfix to be released in v3?

@mheon PTAL

**Comment by rhatdan** [comment]
No plans for that now.

**Comment by mheon** [comment]
Given when this merged, I probably already attempted to backport it to v3.4.2 or v3.4.3, but the patch had cherry-pick conflicts against that branch so I moved on.

**Comment by geraldwuhoo** [comment]
Has this feature been added in v4? Been waiting on this for k3d on podman.

**Comment by Luap99** [comment]
@geraldwuhoo It should be in v4.0

**Pull request merged**